pubsub/producer.go

```diff
@@ -21,18 +22,18 @@ const (
 )
 
 type Marshallable[T any] interface {
-	Marshal() any
-	Unmarshal(val any) (T, error)
+	Marshal() []byte
+	Unmarshal(val []byte) (T, error)
```

Comment:
Your current solution does seem better than most alternatives..
What do you think of this?

* instead of Marshallable - define a "Marshaller" interface with:
```
type Marshaller[T any] interface {
    Marshall(T) []byte
    Unmarshall([]byte]) (T, error)
}
```

* Producer/Consumer are generics over [Request any, Response any]
* NewProducer and NewConsumer functions both accept two new vars of the interface type Marshaller[Request] and Marshaller[Response] - store it in the struct, and use it to marshall/unmarshall

It's a little weird but in a way quite common in go - we use an interface where the exact type doesn't really matter (meaning - we'll never have to convert this interface to a specific type), and a generic where it does.